src/ucp/dt/datatype_iter.inl

```diff
@@ -505,6 +505,10 @@ ucp_datatype_iter_mem_reg(ucp_context_h context, ucp_datatype_iter_t *dt_iter,
                           ucp_md_map_t md_map, unsigned uct_flags,
                           unsigned dt_mask)
 {
+    if (md_map == 0) {
+        return UCS_OK;
+    }
```

Comment:
let's say we remove this branch.
would it cause something to fail, or just "waste" more time to create memh with md_map==0?

Reply:
the latter

Reply:
it's not clear not me why we would register non-contig type. dt_iter already checks the datatype class before registering

Reply:
here - https://github.com/openucx/ucx/blob/master/src/ucp/rndv/proto_rndv.inl#L52

Reply:
wouldn't it go here?
https://github.com/openucx/ucx/blob/master/src/ucp/dt/datatype_iter.inl#L508

Reply:
yes for generic, but no for IOV